src/libsync/propagatedownload.cpp

```diff
@@ -159,9 +159,14 @@ void GETFileJob::slotMetaDataChanged()
 
     int httpStatus = reply()->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
 
-    // Ignore redirects
-    if (httpStatus == 301 || httpStatus == 302 || httpStatus == 303 || httpStatus == 307 || httpStatus == 308)
+    if (httpStatus == 301 || httpStatus == 302 || httpStatus == 303 || httpStatus == 307 || httpStatus == 308 || httpStatus == 401) {
+        // Ignore redirects and authentication failures.
+        // This is handled by AbstractNetworkJob,  so don't let this job error out in this case.
+        bool ok = disconnect(reply(), &QNetworkReply::finished, this, &GETFileJob::slotReadyRead)
+            && disconnect(reply(), &QNetworkReply::readyRead, this, &GETFileJob::slotReadyRead);
```

Comment:
This works because `newReplyHook` reconnects the signals, right? Suggestion:

```
// Redirects and auth failures (oauth token renew) will end up restarting the job. We do not want to process
// further data from the initial request. newReplyHook() will reestablish signal connections for the follow-up
// request.
```

Also, what if there's a 401 that *doesn't* cause a new job to be spawned? Then `finishedSignal()` wouldn't be emitted.

Reply:
Right, I now changed GETFileJob::finished so this does not happen

Reply:
Alright, that looks like it works. What do you think about adding the comment?

Reply:
Comment updated